src/amber/router/pipe/method.cr

```diff
@@ -0,0 +1,21 @@
+module Amber::Pipe
```

Comment:
My thoughts as well. Originally I had just required params to be parsed before route when they were both handlers. But @eliasjpr refactored it to be like this which made sense to me since they were coupled together.

Now it's still parsing params first which is the part that takes a little bit of time. Once we have the params upgrading the method/verb is basically a single line.